docs/rst/tutorials/cloud/wan_edge_cloud_tls/wan_edge_cloud_tls.rst

```diff
@@ -0,0 +1,390 @@
+
+#.  **ROS 2 Device Controller**.
+    This is a ROS 2 application that sends commands to the edge robot.
+    A basic C++ application has been developed for this tutorial that sends publications under the topic on which the ``turtlesim_node`` listens.
+    By means of these publications (commands) from the controller, and the feedback information that the controller receives from the ``turtlesim_node``, it is possible to control this node automatically without the need for user intervention which facilitates the deployment of the scenario at hand.
+    The key feature of the *DDS Router* is that it is easy to configure, allowing to connect different networks with ROS 2 applications without requiring to apply any changes to the developer's software or applications.
+
+#.  **ROS 2 Router / DDS Router**.
+    *eProsima ROS 2 Router*, a.k.a `DDS Router <https://github.com/eProsima/DDS-Router>`_, is an end-user software application that enables the connection of distributed ROS 2 networks (see DDS Router documentation `here <https://eprosima-dds-router.readthedocs.io/en/latest/>`_).
+    That is, ROS 2 nodes such as publishers and subscribers, or clients and services, deployed in one geographic location and using a dedicated local network will be able to communicate with other ROS 2 nodes deployed in different geographic areas on their own dedicated local networks as if they were all on the same network through the use of *DDS Router*.
+
+    This example presents two routers that enable Internet communication:
+
+    * *DDS Router Edge*. This is the DDS Router that is deployed on the edge robot side. This way it is possible for the robot to communicate out-of-the-box with an external server.
+    * *DDS Router Cloud*. It plays the server role in the communication. It will expose a public network address to which the nodes connect to establish communication.
+
+
+Prerequisites
+-------------
+
+This tutorial will require two machines (*Robot 1* and *Cloud Server*) deployed on different networks (*LAN 1* and *Cloud*).
+It is possible to simulate the scenario by deploying everything needed on the same machine and two virtual networks but let's focus on the case of a real deployment.
+
+It is also necessary to have previously installed *Vulcanexus* using one of the following installation methods:
+
+* :ref:`linux_binary_installation`
+* :ref:`linux_source_installation`
+* :ref:`docker_installation`
+
+.. note::
+
+    `OpenSSL <https://www.openssl.org/>`_, required in this tutorial to generate security keys and certificates, is already part of *Vulcanexus* toolset.
+
+.. _tls_configuration:
+
+TLS configuration
+-----------------
+
+Let us first generate the TLS credentials with which *DDS Router* instances will be configured. In this example,
+Elliptic Curve (EC) keys will be generated, and with/without password versions of the commands will be provided.
+
+Certification Authority (CA)
```

Comment:
Specify in which robots/machines these commands must be executed, and which results will be use in which one